waspc/data/Generator/templates/server/src/auth/providers/email/signup.ts

```diff
@@ -0,0 +1,31 @@
+import waspServerConfig from '../../../config.js';
+import { EmailSender, EmailFromField } from "../../../email/core/types.js";
+import { handleRejection } from "../../../utils.js";
+import { createEmailVerificationToken, createUser } from "../../utils.js";
+import { GetVerificationEmailContentFn } from './types.js';
+
+export function getSignupRoute({
+    emailSender,
+    fromField,
+    getVerificationEmailContent,
+}: {
+    emailSender: EmailSender;
+    fromField: EmailFromField;
+    getVerificationEmailContent: GetVerificationEmailContentFn;
+}) {
+    return handleRejection(async (req, res) => {
+        const userFields = req.body || {};
+    
+        const user = await createUser(userFields);
```

Comment:
I worry a bit about using the real user model here with pending/to be verified accounts. I think there may be flows that are not desireable (beyond the ability to just spam the system and clutter the user table).

Reply:
> using the real user model here with pending/to be verified accounts

I can see a scenario where somebody knows my e-mail and create an account with my e-mail.

1. They create an account with "miho@xyz.com"
2. I get an e-mail (spam)
3. I later on want to create my real account
4. The e-mail is taken and i would need to use "reset password" to get the actual password

That sounds inconvenient, true. But it seems like it would work itself out.  

> beyond the ability to just spam the system and clutter the user table

That's also a problem with username & password auth as well.

Reply:
The thrid option might be: don't consider an e-mail taken until verified.

Reply:
TODO: research how others are doing e-mail verification